torchsde/_core/base_sde.py

```diff
@@ -48,117 +48,116 @@ def __init__(self, sde):
                     allow_unused=True
-                )
-                for col_idx in range(g_eval[0].size(-1))
+                )[0]
+                for col_idx in range(g.size(-1))
             ]
-            gdg_jvp_eval = misc.seq_add(*gdg_jvp_eval)
-        return misc.convert_none_to_zeros(gdg_jvp_eval, y)
+            dg_ga_jvp = sum(dg_ga_jvp)
+        return dg_ga_jvp
 
-    def gdg_jvp_column_sum_v2(self, t, y, a):
+    def dg_ga_jvp_column_sum_v2(self, t, y, a):
         # Faster, but more memory intensive.
-        requires_grad = torch.is_grad_enabled()  # BP through solver.
+        requires_grad = torch.is_grad_enabled()
         with torch.enable_grad():
-            y = [y_ if y_.requires_grad else y_.detach().requires_grad_(True) for y_ in y]
-            g_eval = self._base_sde.g(t, y)
-            v = [torch.bmm(g_eval_, a_) for g_eval_, a_ in zip(g_eval, a)]
-
-            batch_size, d, m = g_eval[0].size()  # TODO: Relax this assumption.
-            y_dup = [torch.repeat_interleave(y_, repeats=m, dim=0) for y_ in y]
-            g_eval_dup = self._base_sde.g(t, y_dup)
-            v_flat = [v_.transpose(1, 2).flatten(0, 1) for v_ in v]
-            gdg_jvp_eval = misc.jvp(
-                g_eval_dup, y_dup, grad_inputs=v_flat, create_graph=requires_grad, allow_unused=True
+            y = y if y.requires_grad else y.detach().requires_grad_(True)
+            g = self._base_sde.g(t, y)
+            ga = torch.bmm(g, a)
+
+            batch_size, d, m = g.size()
+            y_dup = torch.repeat_interleave(y, repeats=m, dim=0)
+            g_dup = self._base_sde.g(t, y_dup)
+            ga_flat = ga.transpose(1, 2).flatten(0, 1)
+            dg_ga_jvp, = misc.jvp(
+                outputs=g_dup,
+                inputs=y_dup,
+                grad_inputs=ga_flat,
+                create_graph=requires_grad,
+                allow_unused=True
             )
-            gdg_jvp_eval = misc.convert_none_to_zeros(gdg_jvp_eval, y)
-            gdg_jvp_eval = [t.reshape(batch_size, m, d, m).permute(0, 2, 1, 3) for t in gdg_jvp_eval]
-            gdg_jvp_eval = [t.diagonal(dim1=-2, dim2=-1).sum(-1) for t in gdg_jvp_eval]
-        return gdg_jvp_eval
-
-    def _skip(self, t, y, v):  # noqa
-        return [0.] * len(y)
-
-
-class TupleSDE(BaseSDE):
-
-    def __init__(self, sde):
-        super(TupleSDE, self).__init__(noise_type=sde.noise_type, sde_type=sde.sde_type)
-        self._base_sde = sde
-
-    def f(self, t, y):
-        return self._base_sde.f(t, y[0]),
-
-    def g(self, t, y):
-        return self._base_sde.g(t, y[0]),
+            dg_ga_jvp = dg_ga_jvp.reshape(batch_size, m, d, m).permute(0, 2, 1, 3)
+            dg_ga_jvp = dg_ga_jvp.diagonal(dim1=-2, dim2=-1).sum(-1)
+        return dg_ga_jvp
 
-    def h(self, t, y):
-        return self._base_sde.h(t, y[0]),
+    def _return_zero(self, t, y, v):  # noqa
+        return 0.
 
 
 class RenameMethodsSDE(BaseSDE):
```

Comment:
This doesn't need `h` any more.

Reply:
There's still an unused prior_drift argument.

Reply:
Done.